backend/entityservice/tasks/encoding_uploading.py

```diff
@@ -56,12 +60,21 @@ def pull_external_data(project_id, dp_id,
     bucket_name = encoding_object_info['bucket']
     object_name = encoding_object_info['path']
 
-    stat, encodings_stream = stat_and_stream_object(bucket_name, object_name, parse_minio_credentials(encoding_credentials))
+    stat, encodings_stream = stat_and_stream_object(bucket_name, object_name, env_credentials)
     count = int(stat.metadata['X-Amz-Meta-Hash-Count'])
     size = int(stat.metadata['X-Amz-Meta-Hash-Size'])
     log.debug(f"Processing {count} encodings of size {size}")
     assert count == len(encoding_to_block_map), f"Expected {count} encodings in blocks got {len(encoding_to_block_map)}"
 
+    # load clks properly
+    encodings = json.loads(encodings_stream.data.decode())['clks']
```

Comment:
consider using `ijson`. It allows you to iteratively read json and thus avoids having to load the whole file into memory.